src/LinearSolverTypes_Multigrid.f90

```diff
@@ -0,0 +1,973 @@
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+!                          Futility Development Group                          !
+!                             All rights reserved.                             !
+!                                                                              !
+! Futility is a jointly-maintained, open-source project between the University !
+! of Michigan and Oak Ridge National Laboratory.  The copyright and license    !
+! can be found in LICENSE.txt in the head directory of this repository.        !
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+!> @brief Module provides a linear system type and methods to solve systems
+!> of equations via a multigrid method
+!>
+!> To use LinearSolverType_Multigrid:
+!>   (1) Initialize the linear solver using %init
+!>   (2) Fill out MultigridMesh appropriately.
+!>   (3) Call LS%fillInterpMats
+!>   (4) Call LS%setupPETScMG
+!>
+!> Step 3 allocates and fills out all of the interpolation operators in PETSc's
+!>   MG structure.  The Galerkin process is assumed so coarse grid operators
+!>   are generated as a triple product of the form RAI, where I is interp.,
+!>   R is restriction (transport of interpolation), and A is the fine grid
+!>   matrix from the next grid up.
+!> Step 4 sets up the smoother options on each level and misc. details for MG
+!>
+!> For more information, see:
+!>  http://www.mcs.anl.gov/petsc/petsc-3.6/docs/manualpages/PC/PCMG.html
+!>
+!> For valid reference lists
+!> see @ref MatrixTypes::LinearSolverTypes_Declare_ValidParams
+!> "LinearSolverTypes_Declare_ValidParams".
+!>
+!> Currently supported TPLs include:
+!>  - PETSc (with interfaces to KSP)
+!>
+!> @par Module Dependencies
+!>  - @ref IntrType "IntrType": @copybrief IntrType
+!>  - @ref BLAS "BLAS": @copybrief BLAS
+!>  - @ref Times "Times": @copybrief Times
+!>  - @ref ExceptionHandler "ExceptionHandler": @copybrief ExceptionHandler
+!>  - @ref Allocs "Allocs": @copybrief Allocs
+!>  - @ref ParameterLists "ParameterLists": @copybrief ParameterLists
+!>  - @ref ParallelEnv "ParallelEnv": @copybrief ParallelEnv
+!>  - @ref VectorTypes "VectorTypes": @copybrief VectorTypes
+!>  - @ref MatrixTypes "MatrixTypes": @copybrief MatrixTypes
+!>  - @ref LinearSolverTypes "LinearSolverTypes": @copybrief LinearSolverTypes
+!>
+!> @par EXAMPLES
+!> @code
+!>
+!> @endcode
```

Comment:
Either fill in or just delete this block.  It's not really accomplishing anything as is.

Reply:
Good catch, thanks!